app/controllers/gift_requests_controller.rb

```diff
@@ -7,20 +7,23 @@ class GiftRequestsController < ApplicationController
 
   def new
     @back_route = wishlist_path(params[:id])
-
     @gift_request = GiftRequest.new
-    render "gift_request", locals: {list_id: params[:id]}
+    @list_name = List.find(params[:id]).name
+
+    render "gift_request"
   end
 
   def edit
     @back_route = wishlist_path(params[:id])
-
-    render "gift_request", locals: {list_id: params[:id]}
+    
+    render "gift_request"
```

Comment:
Mmm I think this should be ok. In `gift_request.slim` I have:

`"#{@list_name.presence || @gift_request.list.name}"`

`@list_name` will be defined for the create action and for the edit action I have derive it from the `@gift_request` instance. Let me know if that makes sense or if I'm missing something.